features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/featureflags/internal/model/DatadogEvaluationContextTest.kt

```diff
@@ -0,0 +1,316 @@
+
+        // When
+        val datadogContext = DatadogEvaluationContext.from(publicContext, mockInternalLogger)
+
+        // Then
+        assertThat(datadogContext?.targetingKey).isEqualTo(targetingKey)
+        assertThat(datadogContext?.attributes).isEmpty()
+        verifyNoInteractions(mockInternalLogger)
+    }
+
+    @Test
+    fun `M stringifies numbers correctly W from() { different number types }`(forge: Forge) {
+        // Given
+        val targetingKey = forge.anAlphabeticalString()
+        val intValue = forge.anInt()
+        val doubleValue = forge.aDouble()
+        val longValue = forge.aLong()
+        val floatValue = forge.aFloat()
+
+        val publicContext = EvaluationContext(
+            targetingKey = targetingKey,
+            attributes = mapOf(
+                "intAttr" to intValue,
+                "doubleAttr" to doubleValue,
+                "longAttr" to longValue,
+                "floatAttr" to floatValue
+            )
+        )
+
+        // When
+        val datadogContext = DatadogEvaluationContext.from(publicContext, mockInternalLogger)
+
+        // Then
+        assertThat(datadogContext).isNotNull
+        assertThat(datadogContext?.attributes).isEqualTo(
+            mapOf(
+                "intAttr" to intValue.toString(),
+                "doubleAttr" to doubleValue.toString(),
+                "longAttr" to longValue.toString(),
+                "floatAttr" to floatValue.toString()
+            )
+        )
+        verifyNoInteractions(mockInternalLogger)
+    }
+
+    @Test
+    fun `M return null W from() { blank targeting key }`() {
+        // Given
+        val blankTargetingKey = ""
+        val publicContext = EvaluationContext(blankTargetingKey, emptyMap())
+
+        // When
+        val result = DatadogEvaluationContext.from(publicContext, mockInternalLogger)
+
+        // Then
+        assertThat(result).isNull()
+    }
+
+    @Test
+    fun `M return null W from() { whitespace-only targeting key }`() {
+        // Given
+        val whitespaceTargetingKey = "   "
+        val publicContext = EvaluationContext(whitespaceTargetingKey, emptyMap())
+
+        // When
+        val result = DatadogEvaluationContext.from(publicContext, mockInternalLogger)
+
+        // Then
+        assertThat(result).isNull()
+    }
+
+    @Test
+    fun `M return null W from() { tab-only targeting key }`() {
+        // Given
+        val tabTargetingKey = "\t\t"
+        val publicContext = EvaluationContext(tabTargetingKey, emptyMap())
+
+        // When
+        val result = DatadogEvaluationContext.from(publicContext, mockInternalLogger)
+
+        // Then
+        assertThat(result).isNull()
+    }
+
+    @Test
+    fun `M return true W isValid() { valid context }`() {
+        // Given
+        val context = DatadogEvaluationContext("user123", mapOf("plan" to "premium"))
+
+        // When
+        val isValid = context?.isValid()
```

Comment:
`context` is not `null` here

```suggestion
        val isValid = context.isValid()
```